include/zephyr/rtio/rtio.h

```diff
@@ -774,6 +921,61 @@ static inline void rtio_cqe_submit(struct rtio *r, int result, void *userdata)
 #endif
 }
 
+static inline struct rtio_sqe *rtio_sqe_handle_mempool(struct rtio *r, struct rtio_sqe *sqe)
+{
+#ifndef CONFIG_RTIO_SYS_MEM_BLOCKS
+	UNUSED_ARG(r);
+#else
+	__ASSERT_NO_MSG(r != NULL);
+	__ASSERT_NO_MSG(sqe != NULL);
+
+	if (sqe->op != RTIO_OP_RX || sqe->buf != NULL || r->mempool == NULL) {
+		return sqe;
+	}
+
+//	if (sys_mem_blocks_alloc(r->mempool, 1, (void**)&sqe->buf)) {
+//		/* Failed to allocate, return sqe and let the consumer decide what to do */
+//		return sqe;
+//	}
+//
+//	sqe->buf_len = r->mempool_blk_size;
+	sqe->flags |= RTIO_SQE_MEMPOOL_BUFFER;
+	sqe->mempool = r->mempool;
+	sqe->mempool_blk_size = r->mempool_blk_size;
+#endif
+	return sqe;
+}
+
+static inline int rtio_sqe_rx_buf(const struct rtio_sqe *sqe, uint8_t **buf, uint32_t *buf_len)
```

Comment:
If you take the iodev_sqe you have the rtio context and sqe, you can directly allocate from the mempool then if the flag is set or return the sqe provided buffer pointer and len